scripts/packages/VSCodeNotebookServer/src/stdio.jl

```diff
@@ -0,0 +1,277 @@
+    return io
+end
+
+# # logging in verbose mode goes to original stdio streams.  Use macros
+# # so that we do not even evaluate the arguments in no-verbose modes
+
+# using Printf
+# function get_log_preface()
+#     t = now()
+#     taskname = get(task_local_storage(), :IJulia_task, "")
+#     @sprintf("%02d:%02d:%02d(%s): ", Dates.hour(t),Dates.minute(t),Dates.second(t),taskname)
+# end
+
+# macro vprintln(x...)
+#     quote
+#         if verbose::Bool
+#             println(orig_stdout[], get_log_preface(), $(map(esc, x)...))
+#         end
+#     end
+# end
+
+# macro verror_show(e, bt)
+#     quote
+#         if verbose::Bool
+#             showerror(orig_stderr[], $(esc(e)), $(esc(bt)))
+#         end
+#     end
+# end
+
+# name=>iobuffer for each stream ("stdout","stderr") so they can be sent in flush
+const bufs = Dict{String,IOBuffer}()
+const stream_interval = 0.1
+# maximum number of bytes in libuv/os buffer before emptying
+const max_bytes = 10 * 1024
+# max output per code cell is 512 kb by default
+const max_output_per_request = Ref(1 << 19)
+
+"""
+Continually read from (size limited) Libuv/OS buffer into an `IObuffer` to avoid problems when
+the Libuv/OS buffer gets full (https://github.com/JuliaLang/julia/issues/8789). Send data immediately
+when buffer contains more than `max_bytes` bytes. Otherwise, if data is available it will be sent every
+`stream_interval` seconds (see the Timers set up in watch_stdio). Truncate the output to `max_output_per_request`
+bytes per execution request since excessive output can bring browsers to a grinding halt.
+"""
+function watch_stream(rd::IO, name::AbstractString)
+    task_local_storage(:IJulia_task, "read $name task")
+    try
+        buf = IOBuffer()
+        bufs[name] = buf
+        while !eof(rd) # blocks until something is available
+            nb = bytesavailable(rd)
+            if nb > 0
+                stdio_bytes[] += nb
+                # if this stream has surpassed the maximum output limit then ignore future bytes
+                if stdio_bytes[] >= max_output_per_request[]
+                    read(rd, nb) # read from libuv/os buffer and discard
+                    if stdio_bytes[] - nb < max_output_per_request[]
+                        JSONRPC.send_notification(conn_endpoint[], "streamoutput", Dict{String,Any}("name" => "stderr", "current_request_id" => current_request_id[], "data" => "Excessive output truncated after $(stdio_bytes[]) bytes."))
+                    end
+                else
+                    write(buf, read(rd, nb))
+                end
+            end
+            if buf.size > 0
+                if buf.size >= max_bytes
+                    # send immediately
+                    send_stream(name)
+                end
+            end
+        end
+    catch e
+        # the IPython manager may send us a SIGINT if the user
+        # chooses to interrupt the kernel; don't crash on this
+        if isa(e, InterruptException)
```

Comment:
Based on my knowledge of interrupts on windows, SIGINT doesn't work on windows.
I believe messages are sent to interrupt kernels.
Please could you confirm, and can we add a message in the Julia kernel for this if thats the case. 
FYI - We can do that later, and I could just create a separate issue to track such requirements

Reply:
Yes, we need to add another message for that. We have that for the REPL kernel already, so I think that should be relatively straightforward.

Reply:
I'll create an issue to track that requirement, to ensure we don't forget.